pipeline-scripts/release.groovy

```diff
@@ -21,24 +23,64 @@ def stageValidation(quay_url, name, advisory) {
         error("Payload ${name} already exists! Cannot continue.")
     }
 
-    // AMH - This may be optional?
-    if (advisory != "") {
-        echo "Verifying advisory exists"
+    if (!advisory) {
+        echo "Getting current advisory for OCP $version from build data..."
         res = commonlib.shell(
                 returnAll: true,
-                script: "elliott get ${advisory}"
-        )
+                script: "elliott --group=openshift-${version} get --json - --use-default-advisory image",
+            )
+        if(res.returnStatus != 0) {
+            error("🚫 Advisory number for OCP $version couldn't be found from ocp_build_data.")
+        }
+    } else {
+        echo "Verifying advisory ${advisory} exists"
+        res = commonlib.shell(
+                returnAll: true,
+                script: "elliott --group=openshift-${version} get --json - -- ${advisory}",
+            )
 
         if(res.returnStatus != 0){
             error("Advisory ${advisory} does not exist! Cannot continue.")
         }
     }
+
+    def advisoryInfo = readJSON text: res.stdout
+    retval.advisoryInfo = advisoryInfo
+
+    echo "Verifying advisory ${advisoryInfo.id} (https://errata.engineering.redhat.com/advisory/${advisoryInfo.id}) status"
+    if (advisoryInfo.status != 'QE') {
+        error("🚫 Advisory ${advisoryInfo.id} is not in QE state.")
+    }
+    echo "✅ Advisory ${advisoryInfo.id} is in QE state."
+
+    // Extract live ID from advisory info
+    // Examples:
+    // - advisory with live ID:
+    //     "errata_id": 2681,
+    //     "fulladvisory": "RHBA-2019:2681-02",
+    //     "id": 46049,
+    //     "old_advisory": "RHBA-2019:46049-02",
+    // - advisory without:
+    //     "errata_id": 46143,
+    //     "fulladvisory": "RHBA-2019:46143-01",
+    //     "id": 46143,
+    //     "old_advisory": null,
+    if (advisoryInfo.errata_id != advisoryInfo.id && advisoryInfo.fulladvisory && advisoryInfo.old_advisory) {
```

Comment:
optional nitpick:
to extract this complex conditional into a function with a meaningful name, something like:
```groovy
def hasLiveId(advisoryInfo) {
    return advisoryInfo.errata_id != advisoryInfo.id && advisoryInfo.fulladvisory && advisoryInfo.old_advisory
}

// ...

if (hasLiveID(advisoryInfo)) {
    // ....
}
```

Reply:
Maybe I can move it to commonlib?